synapse/async.py

```diff
@@ -1,14 +1,5 @@
-import time
-import traceback
-import threading
-import collections
```

Comment:
Yep - it took me a minute to figure out where the `time` module was coming from. I'd rather explicitly list them.